app/models/rule_group_rule.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# Join table with the has-many-belongs-to-many relations between
+# RuleGroup and Rule
+class RuleGroupRule < ApplicationRecord
+  # These keep track of the parent-child relationship between
+  # rules and rule_groups
+  belongs_to :rule, optional: true
+  belongs_to :rule_group, optional: true
+  # These keep track of required/conflicting rules and rule groups
+  belongs_to :depending, polymorphic: true, optional: true
+  belongs_to :requiring, polymorphic: true, optional: true
+  belongs_to :conflicting, polymorphic: true, optional: true
```

Comment:
Okay, the `depending` is definitely a wrong nam**ing**, it's confus**ing** and IMO it should not be end**ing** with -**ing** so it's not mistaken as someth**ing** similar to the other 2 fields as it is not.

Sorry :see_no_evil: 

> Voila! In view humble vaudevillian veteran, cast vicariously as both victim and villain by the vicissitudes of fate. This visage, no mere veneer of vanity, is a vestige of the “vox populi” now vacant, vanished. However, this valorous visitation of a bygone vexation stands vivified, and has vowed to vanquish these venal and virulent vermin, van guarding vice and vouchsafing the violently vicious and voracious violation of volition. The only verdict is vengeance; a vendetta, held as a votive not in vain, for the value and veracity of such shall one day vindicate the vigilant and the virtuous. Verily this vichyssoise of verbiage veers most verbose, so let me simply add that it’s my very good honour to meet you and you may call me V.

Reply:
Also we're not really us**ing** the full potential of the DB here as it seems like you're stor**ing** the conflict**ing** and requir**ing** separately. For example if you have rule A conflict**ing** with rule B and requir**ing** rule C, you'd need only a single DB row, but as far as I understand your code below, it is mak**ing** two separate ones.

Let's do a quick meet**ing** about these when you're online.

Reply:
Right, I thought `depending` sounded a little weird but I named it that since it's a polymorphic column but I don't need to follow the naming conventions because I don't think it's necessary. I could rename it just `dependent`? Because it's dependent on something else when it `requires` or `conflicts` with a rule or rule group?

The reason I wasn't storing them in the same column is because a rule or a rule group can conflict with or require 0-n rules or rule groups so for example, if a rule conflicts with 2 different rule groups, wouldn't that need to be 2 different rows? And if it also requires a rule/rule group, I'm not sure how to store that all in one row